releasenotes/notes/route-collapse.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+issue:
+- 28659
+releaseNotes:
+- |
+  **Optimized** generated routing configuration to merge virtual hosts with the same routing configuration. This improves performance for Virtual Services with multiple hostnames defined.
+
+upgradeNotes:
+- title: EnvoyFilter `match.routeConfiguration.vhost.name` semantics change
```

Comment:
Maybe `semantics have changed` I'm not sure the is is any better, and could leave it as well.